### ISSUE TITLE
`minimize_dispersion()` returns mapping that displaces all population

### ISSUE THREAD

**Issue by gabeschoenbach** [mggg/plan-evaluation-processing#11]
I'm using the [Wisconsin block shapefile](http://data.mggg.org.s3-website.us-east-2.amazonaws.com/census-2020/wi/) and loading in the `SB622` proposed Congressional plan.

```python
gdf = gpd.read_file("shapes/wi_block/")

benchmark = pd.read_csv("/Users/gabe/Dropbox/PlanAnalysis/proposed_plans/Wisconsin/congress/benchmark-blocks.csv", dtype=str)
benchmark.columns = ["GEOID20", "Benchmark"]
SB = pd.read_csv("/Users/gabe/Dropbox/PlanAnalysis/proposed_plans/Wisconsin/congress/SB622-blocks.csv", dtype=str)
SB.columns = ["GEOID20", "SB622"]
gdf = gdf.merge(benchmark, on="GEOID20")
gdf = gdf.merge(SB, on="GEOID20")
```

If we calculate the dispersion of `SB622` compared to the `Benchmark` plan, the one used in the 2010 cycle, we get 384,456.

```python
calculate_dispersion(gdf, "Benchmark", "SB622", "TOTPOP20")
```

Then, we can use `minimize_dispersion()` to get a minimal mapping and call it `Minimal`. However when we check the dispersion of `Minimal` we see that it displaces all 5,893,718 of the `TOTPOP20`. Maybe I'm using this wrong somehow??

```python
minimal_mapping = minimize_dispersion(gdf, "Benchmark", "SB622", "TOTPOP20")
gdf["Minimal"] = gdf.SB622.apply(lambda x:minimal_mapping[int(x)])
calculate_dispersion(gdf, "Benchmark", "Minimal", "TOTPOP20")
```

**Comment by InnovativeInventor** [comment]
Ah, this is a classic Python types issue. If you cast `SB622` and `Benchmark` to ints beforehand, everything works as expected. (minimize_dispersion will return the identity mapping). I am working on a patch that will warn the user when such a types issue occurs.

**Comment by InnovativeInventor** [comment]
Closed due to the merging of #12.
